pkg/cmd/proxy/probe/exec.go

```diff
@@ -0,0 +1,324 @@
+	}
+	// building tls config from secret data
+	restConfig, err := o.ClusteradmFlags.KubectlFactory.ToRESTConfig()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed building cilent config")
+	}
+	nativeClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed building cilent")
+	}
+	caSecret, err := nativeClient.CoreV1().Secrets(proxyConfig.Spec.ProxyServer.Namespace).
+		Get(context.TODO(), inClusterSecretProxyCA, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed getting CA secret")
+	}
+	caData := caSecret.Data["ca.crt"]
+	certSecret, err := nativeClient.CoreV1().Secrets(proxyConfig.Spec.ProxyServer.Namespace).
+		Get(context.TODO(), inClusterSecretClient, metav1.GetOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed getting cert & key secret")
+	}
+	certData := certSecret.Data["tls.crt"]
+	keyData := certSecret.Data["tls.key"]
+
+	// building tls config from pem-encoded data
+	tlsCfg, err := buildTLSConfig(caData, certData, keyData, "127.0.0.1", nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed building TLS config from secret")
+	}
+	return tlsCfg, nil
+
+}
+
+func (o *Options) visit(
+	w *writer,
+	hubRestConfig *rest.Config,
+	addonClient addonv1alpha1client.Interface,
+	dialFunc k8snet.DialFunc,
+	clusterName string) error {
+
+	addon, err := addonClient.AddonV1alpha1().ManagedClusterAddOns(clusterName).
+		Get(context.TODO(), common.AddonName, metav1.GetOptions{})
+	installed := "True"
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "failed getting managed cluster addon for cluster %v", clusterName)
+		}
+		installed = "False"
+	}
+	addonAvailable := "False"
+	for _, cond := range addon.Status.Conditions {
+		if cond.Type == addonv1alpha1.ManagedClusterAddOnConditionAvailable {
+			addonAvailable = string(cond.Status)
+		}
+	}
+
+	copiedCfg := rest.CopyConfig(hubRestConfig)
+	copiedCfg.Dial = dialFunc
+	copiedCfg.TLSClientConfig = rest.TLSClientConfig{}
+	copiedCfg.TLSClientConfig.Insecure = true
+
+	rt, err := rest.TransportFor(copiedCfg)
+	if err != nil {
+		return errors.Wrapf(err, "failed creating roundtripper for cluster %v", clusterName)
+	}
+	req := &http.Request{
+		Method: "GET",
+		Host:   clusterName,
+		URL: &url.URL{
+			Scheme: "https",
+			Host:   clusterName,
+			Path:   "/healthz",
+		},
+	}
+	health := "False"
+	latency := "<none>"
+	start := time.Now()
+	isFailed := false
+	resp, err := rt.RoundTrip(req)
+	if err != nil {
+		isFailed = true
+		klog.Errorf("Failed requesting /healthz endpoint for cluster %v", clusterName)
+		if strings.Contains(err.Error(), "dial timeout") {
+			latency = "<timeout>"
+		}
+	}
+	if !isFailed {
+		end := time.Now()
+		data, _ := ioutil.ReadAll(resp.Body)
+		if "ok" == string(data) {
+			health = "True"
+			latency = end.Sub(start).String()
+		}
+	}
+
+	w.print(clusterName, installed, addonAvailable, health, latency)
```

Comment:
TODO use a common table convertor in the future.

Reply:
added